src/simple_websocket/ws.py

```diff
@@ -348,6 +352,10 @@ def choose_subprotocol(self, request):
                 return subprotocol
         return None
 
+    def close(self, reason=None, message=None):
+        super().close(reason=reason, message=message)
+        self.sock.close()
```

Comment:
Is this the correct way to handle a graceful connection end? It seems the side that is going away sends a close packet, but does not give the other side to acknowledge and respond, since it immediately closes the socket as well. I think the correct closing handshake is for the first party to send the close packet, then wait for the other side to response with their own close packet, and only then both close their sockets.

Reply:
I think you're right, we shouldn't force the socket to close right away, I overlooked that part of the websocket protocol. However, the Client, as far as I can understand, is in fact closing the socket after super().close() (line 440), actually it was bit of "cargo cult" from my part using that pattern on Server.

I can definitely remove this override of close() if you agree it's the best way and test again. However I'm not comfortable changing Client.close() as I'm not confident in my ability to test it fully.

Reply:
Reanimating this: Do you agree on removing this on close() of Server?

Reply:
Yes, as long as the close call happens when the server receives the close confirmation packet from the client. I think that should be already in place, but should probably be verified.

Reply:
Thank you, I'll change it, analyse the code to check for the close on the Server and retest the original problematic case again